File: lib/test/api_test.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:test/test.dart';

void main() {
  test('Test predict_weather route', () async {
    final response = await http.get(Uri.parse('http://192.168.88.38:5000/predict_weather?time=12:00&date=2024-04-01'));

    expect(response.statusCode, equals(200));
    
    final Map<String, dynamic> data = json.decode(response.body);
    expect(data.containsKey('temperature'), equals(true));
    expect(data.containsKey('precipitation'), equals(true));
    expect(data.containsKey('humidity'), equals(true));
  });
}

File: lib/test/api_test.py
import unittest
import requests

class TestAPI(unittest.TestCase):

    def test_predict_weather_route(self):
        url = 'http://192.168.88.38:5000/predict_weather'
        params = {'time': '12:00', 'date': '2024-04-01'}
        response = requests.get(url, params=params)

        self.assertEqual(response.status_code, 200)
        data = response.json()
        self.assertIn('temperature', data)
        self.assertIn('precipitation', data)
        self.assertIn('humidity', data)

if __name__ == '__main__':
    unittest.main()
